expirationd.lua

```diff
@@ -322,6 +325,35 @@ local function expirationd_run_task(name, space_id, is_tuple_expired, options)
     end
     task.process_expired_tuple = options.process_expired_tuple or default_tuple_drop
 
+    -- validate index
+    local expire_index = box.space[space_id].index[0]
+    if options.index then
+        if type(options.index) == "string" then
+            local index_exists = false
+            for _, index in pairs(box.space[space_id].index) do
+                if index.name == options.index then
+                    expire_index = index
+                    index_exists = true
+                    break
+                end
+            end
```

Comment:
```
box.cfg{}
_ = box.schema.space.create('test')
_ = box.space.test:create_index('pk')
_ = box.space.test:create_index('sk')

tarantool> box.space.test.index['pk'] ~= nil
---
- true
...

tarantool> box.space.test.index[1] ~= nil
---
- true
...

tarantool> box.space.test.index['sk'] ~= nil
---
- true
...

tarantool> box.space.test.index['doesnt_exist'] ~= nil
---
- false
...
```

Why do you still use a loop to check that index exists?

Reply:
this is a problem that without thinking I copied and pasted the idea from https://github.com/moonlibs/indexpiration, I thought it makes sense